packages/@react-aria/color/src/useHexColorField.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaHexColorFieldProps} from '@react-types/color';
+import {
+  defaultMaxValue,
+  defaultMinValue,
+  HexColorFieldState,
+  useColor
+} from '@react-stately/color';
+import {
+  HTMLAttributes,
+  LabelHTMLAttributes,
+  RefObject,
+  useEffect
+} from 'react';
+import {mergeProps, useId} from '@react-aria/utils';
+import {useSpinButton} from '@react-aria/spinbutton';
+import {useTextField} from '@react-aria/textfield';
+
+interface HexColorFieldAria {
+  labelProps: LabelHTMLAttributes<HTMLLabelElement>,
+  inputFieldProps: HTMLAttributes<HTMLInputElement>
+}
+
+export function useHexColorField(
+  props: Omit<AriaHexColorFieldProps, 'value' | 'defaultValue' | 'onChange'>,
+  state: HexColorFieldState,
+  ref: RefObject<HTMLInputElement>
+): HexColorFieldAria {
+  let {
+    isDisabled,
```

Comment:
_(ignore the line of code cited; it's just to start this thread)_

Hi @snowystinger, regarding
> for some reason i can still select the text in the disabled control

I patterned `HexColorField.ts` from `SearchField.ts`, and I noticed you can also select text even when the input is disabled there. I'll go ahead and rework the disabled state for this component but `SearchField` behavior might be worth looking into as well

Reply:
<img width="566" alt="Screen Shot 2020-09-29 at 2 04 27 PM" src="https://user-images.githubusercontent.com/5743833/94615492-ba40fc80-025c-11eb-888b-a367d85a9f31.png">

Reply:
<img width="580" alt="Screen Shot 2020-09-29 at 2 05 10 PM" src="https://user-images.githubusercontent.com/5743833/94615573-d775cb00-025c-11eb-8c62-0daa997e96c3.png">

Reply:
oo, thanks, i only compared it to textfield, i wonder what the difference is

Reply:
Hi @snowystinger, I re-examined the disabled behavior for `TextField` and it turns out the text in its `isDisabled` story is not selectable because it's only a placeholder and has no value 😆. Text is selectable on a disabled `TextField`

![Screenshot 2020-10-01 164456](https://user-images.githubusercontent.com/5743833/94873946-763d2b80-0405-11eb-8e16-1d822e00765e.png)

Reply:
Resolving this since behavior is consistent with other components

Reply:
HAHAHAHA, awesome, thanks for figuring that out